bleak/backends/bluezdbus/manager.py

```diff
@@ -582,7 +582,10 @@ def remove_device_watcher(self, watcher: DeviceWatcher) -> None:
             The device watcher token that was returned by
             :meth:`add_device_watcher`.
         """
-        self._device_watchers.remove(watcher)
+        device_path = watcher.device_path
+        self._device_watchers[device_path].remove(watcher)
+        if not self._device_watchers[device_path]:
+            del self._device_watchers[device_path]
```

Comment:
Probably doesn't hurt to leave the list since devices are likely to be seen again (you're probably the only person that has 1000s of devices around :laughing:)

Reply:
I was a bit worried about devices with random mac addresses would build up and leak memory over time

Reply:
Ah, didn't consider that case.